Add IO.sequence

regiskuckaertz opened the pull request:
Addresses https://github.com/scalaz/scalaz-zio/issues/10

jdegoes commented on core/jvm/src/test/scala/scalaz/zio/RTSSpec.scala:
```diff
@@ -333,7 +333,7 @@ class RTSSpec(implicit ee: ExecutionEnv) extends Specification with AroundTimeou
 
   def testRaceAllOfValues =
-    unsafePerformIO(IO.raceAll[Int, Int](List(IO.fail(42), IO.now(24))).attempt) == Right(24)
+    unsafePerformIO(IO.raceAll[Int, Int](List(IO.fail(42).delay(1.second), IO.now(24))).attempt) == Right(24)
```
`race` should only return the first winner. If this is not working, it's a bug in `race` and needs to be fixed. Just add a pending to the test so we can know to fix it and we can file a separate bug.

regiskuckaertz replied:
Without the delay this test succeeds from time to time, is there a way to avoid specs2 reporting the test is fixed and can be enabled? It kind of defeats the purpose.

ktonga replied:
Add `.pendingUntilFixed` to the offending test case

regiskuckaertz commented:
Yeah I did that, then specs2 failed with the message: This test is now
fixed. I've updated as you asked but I'm afraid it will fail in Travis.

On Wed, 13 Jun 2018 at 16:06 Gastón Tonietti <notifications@github.com>
wrote:

> *@ktonga* commented on this pull request.
> ------------------------------
>
> In core/jvm/src/test/scala/scalaz/zio/RTSSpec.scala
> <https://github.com/scalaz/scalaz-zio/pull/39#discussion_r195120263>:
>
> > @@ -333,7 +333,7 @@ class RTSSpec(implicit ee: ExecutionEnv) extends Specification with AroundTimeou
>      unsafePerformIO(IO.point(42).race(IO.never[Throwable, Int])) == 42
>
>    def testRaceAllOfValues =
> -    unsafePerformIO(IO.raceAll[Int, Int](List(IO.fail(42), IO.now(24))).attempt) == Right(24)
> +    unsafePerformIO(IO.raceAll[Int, Int](List(IO.fail(42).delay(1.second), IO.now(24))).attempt) == Right(24)
>
> Add .pendingUntilFixed to the offending test case
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/scalaz/scalaz-zio/pull/39#discussion_r195120263>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAmc2M-krG6DBUnjB_q3dibzslN-TVjBks5t8SpygaJpZM4Uk-E2>
> .
>

-- 

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 

Guardian News & Media Limited is a member of Guardian Media Group plc. 
Registered Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP. 
 Registered in England Number 908396

tkroman commented on core/shared/src/main/scala/scalaz/zio/IO.scala:
```diff
@@ -796,6 +796,15 @@ object IO {
       .map(_.result())
 
+  /**
+   * Evaluate each effet in the structure from left to right, and collect
```
effe_c_t :)

regiskuckaertz commented:
@jdegoes I believe this is as ready as it can get

jdegoes commented:
@regiskuckaertz Thanks for the contribution!
